addons/l10n_it_edi_sdi_coop/models/account_edi_format.py

```diff
@@ -0,0 +1,268 @@
+# -*- coding:utf-8 -*-
+# Part of Odoo. See LICENSE file for full copyright and licensing details.
+
+from odoo import models, _
+
+from lxml import etree
+import base64
+from cryptography.hazmat.backends import default_backend
+from cryptography.hazmat.primitives import serialization
+from cryptography.hazmat.primitives import hashes
+from cryptography.hazmat.primitives.asymmetric import padding
+from cryptography.fernet import Fernet
+
+
+import logging
+
+_logger = logging.getLogger(__name__)
+
+
+class AccountEdiFormat(models.Model):
+    _inherit = 'account.edi.format'
+
+    # -------------------------------------------------------------------------
+    # Helpers
+    # -------------------------------------------------------------------------
+
+    def _l10n_it_edi_get_proxy_user(self, company):
+        return self.env['account_edi_proxy_client.user'].search([
+            ('company_id', '=', company.id),
+            ('edi_identification', '=', company.l10n_it_codice_fiscale)], limit=1)
+
+    def _l10n_it_edi_decrypt_data(self, data, key, proxy_user):
+        private_key = serialization.load_pem_private_key(
+            base64.b64decode(proxy_user.private_key),
+            password=None,
+            backend=default_backend())
+        key = private_key.decrypt(
+            base64.b64decode(key),
+            padding.OAEP(
+                mgf=padding.MGF1(algorithm=hashes.SHA256()),
+                algorithm=hashes.SHA256(),
+                label=None
+            )
+        )
+        f = Fernet(key)
+        return f.decrypt(base64.b64decode(data))
+
+    # -------------------------------------------------------------------------
+    # Import
+    # -------------------------------------------------------------------------
+
+    def _cron_receive_fattura_pa(self):
+        ''' Check the proxy for incoming invoices.
+        '''
+        companies = self.env['res.company'].search([('l10n_it_codice_fiscale', '!=', False)])
+        for company in companies:
+            proxy_user = self._l10n_it_edi_get_proxy_user(company)
+            if not proxy_user:
+                continue
+            res = proxy_user.make_request('/iap/l10n_it_edi/1/in/RicezioneInvoice', company.l10n_it_edi_test_mode,
+                                          params={'recipient_codice_fiscale': company.l10n_it_codice_fiscale})
+
+            to_ack = []
+            for id_transaction, fattura in res.items():
+                if self.env['ir.attachment'].search([('name', '=', fattura['filename'])], limit=1):
+                    # name should be unique, the invoice already exists
+                    _logger.info('E-invoice already exist: %s', fattura['filename'])
+                    to_ack.append(id_transaction)
+                    continue
+
+                file = self._l10n_it_edi_decrypt_data(fattura['file'], fattura['key'], proxy_user)
+
+                try:
+                    tree = etree.fromstring(file)
```

Comment:
In ir_mail_server. I could play around and extract the code that I'm interested in a new method in account_edi_format and call it from both places, but ir_mail_server is supposed to be deleted in master since we don't want to maintain pec mail anymore, so do you think it's worth it ?